Port from Chicken 4 to 5

alicemaz opened the pull request.

dleslie commented:
I'll have to make sure this still works with chicken 4

dleslie commented:
@alicemaz unless you've done so?

dleslie commented:
Thanks for the port, btw

alicemaz commented:
it almost certainly doesn't. I've never used chicken 4 but afaik there isn't a reasonable (ie, without resorting to cpp or macros to mutate the import list itself) way to support both, the stdlib namespace was reorganized completely

I suppose what to do then is more of a social/procedural question... I assumed most library maintainers would just say "xyz is the last release for the old version of chicken, here is the new release for the current version of chicken" and leave it at that. the main thing is just to have a working version for 4 in the legacy egg index and a working version for 5 in the current one. is there a need for them to be the same?

dleslie commented:
That's sort of what I've done in the past, IE:
https://lists.nongnu.org/archive/html/chicken-users/2019-02/msg00012.html

dleslie commented on monad.release-info:
```diff
@@ -1,6 +0,0 @@
-(repo git "git@github.com:dleslie/monad-egg.git")
```
Please un-remove the release-info file; this should remain for chicken 4 users.
We can create a chicken5.release-info file for the coop to reference.

dleslie commented:
And btw, thanks! It's nice to know someone desired to port this to Chicken5. It's been on my TODO list since C5 came out, but that list is _long_. ;)

alicemaz commented:
ah, ok! I'll make the requested changes. this is actually the first time I've had to interact with the egg system as a contributor at all aha, so I'm still fairly unfamiliar with things. (I actually started using chicken like, a month or two ago; you might guess from my interest in this repo that I'm coming from haskell land)

alicemaz commented:
I've restored the release info for chicken 4, added a new file for chicken 5, bumped the version of this project to 5.0, noted such in the readme and added myself to same

dleslie commented:
Thanks! I'll get to it soon. Busy few days.
